### ISSUE TITLE
Docker-compose without swarm

### ISSUE THREAD

**Issue by theoden-dd** [renskiy/fabricio#190]
Thank you for the tool, Rinat.

Why doesn't it support a case of using docker-compose file on a server, but without swarm?

It might be useful with https://github.com/wemake-services/wemake-django-template for a single node scenario.

If the case is possible, please, describe how it could be configured.

**Comment by renskiy** [comment]
Hi, Denis. Thank you for appreciate. I din't test Fabricio with docker-compose on the server side. But in theory it should work without swarm as well. Hope I will spend some time to check it this weekend.

**Comment by renskiy** [comment]
I have looked to the possibility of using docker compose files outside of using docker stacks (without swarm enabled). And it is not possible right now.

Could you explain your case? So I can decide whether or not this feature should be implemented in the future. Particularly I'm interested why you don't mind to use docker compose but don't want to enable swarm.

**Comment by theoden-dd** [comment]
Thank you for checking the possibility. Here is my case.

First of all, I don't know swarm and not sure that it's a must on the "small" project stage (currently). However I've got a working local compose environment with the [template I mentioned](https://github.com/wemake-services/wemake-django-template). So I want to transfer the compose environment to my yet simple single-node dev and prod servers.

To sum up:

- This is a legacy project and there are a lot of fires here and there and with migrating it to a new stack as a whole. That's why I'm not sure that learning the swarm mode **right now** is worth its complexity.
- While using fabricio from the starting single-node setup with compose could help in future: when we switch to the docker swarm multi-node mode eventually.

**Comment by renskiy** [comment]
swarm mode does not require multi-node configuration. It can be run on a single node. The difference between swarm-enabled docker and just docker is quite small from the user perspective.

Full support of docker compose on the server side is quite impossible due to inability to build images on the remote side (so it may become blocker if rely on this). However you could enable swarm mode on some test node and try to use Fabricio `docker.Stack`. Just make sure your docker compose configuration does not contain any image build stage.

**Comment by renskiy** [comment]
Also you may find interesting [this example](https://github.com/renskiy/fabricio/tree/master/examples/apps/django) which uses Fabricio to deploy Django-based applications.

**Comment by theoden-dd** [comment]
I supposed that it doesn't require multi-node, just thought of it as an over-engineering at the moment. Maybe I'm wrong with that.

I'm not sure if docker compose support is impossible. Actually we don't need to build images. We only need to pull them from a private docker registry. They are pushed there as a result of a pipeline for master/release branches.

Thank you for the example and constructive discussion. Basically I'll stick to 

- ssh to server
- pull containers from registry and compose-file from repo
- docker compose up

at the moment. Something like [another example](https://github.com/toert/django-gitlab-ci-guide/blob/master/devops/deploy.sh). I'll certainly go back to fabricio while switching to swarm. I think we could close the issue.
